sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -388,19 +388,19 @@ static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
 
 	/* dmic */
 	for (i = 1; i <= dmic_num; i++) {
-		links[id].name = devm_kasprintf(dev, GFP_KERNEL,
-						"dmic%02d", i);
-		if (!links[id].name)
-			goto devm_err;
+		if (i == 1) {
+			links[id].name = "dmic01";
+			links[id].cpus->dai_name = "DMIC01 Pin";
+		} else if (i == 2) {
+			links[id].name = "dmic16k";
+			links[id].cpus->dai_name = "DMIC16k Pin";
```

Comment:
@keyonjie what happens when dmic_num is > 2?

Reply:
it can't be > 2 because it's really dai links, not dmics.

Reply:
@plbossart ahh yes, then we dont really need an else here. I think it makes to define these names in a common header file that can be used across machine drivers

Reply:
let me change it.